src/DeviceInput/deviceInputSystem.ts

```diff
@@ -14,9 +14,19 @@ declare const _native: any;
  */
 export class DeviceInputSystem implements IDisposable {
     /**
-     * Callback to be triggered when a device is connected
+     * Returns onDeviceConnected callback property
+     * @returns Callback with function to execute when a device is connected
      */
-    public onDeviceConnected: (deviceType: DeviceType, deviceSlot: number) => void = () => { };
+    public get onDeviceConnected() { return this._onDeviceConnected; }
+
+    /**
+     * Sets callback function when a device is connected
+     * @param callback Function to execute when a device is connected
+     */
+    public set onDeviceConnected(callback) {
+        this._onDeviceConnected = callback;
+        this._checkForConnectedDevices();

```

Comment:
What happens if the gamepad connection events fire between the constructor and someone setting this callback?

What about keyboard and mouse? Do these fire device connected?

Reply:
If a gamepadconnected event fires before the callback is set, what _should_ happen is that the device will still be added to the DeviceInputSystem by the eventlistener body.  This is the same for the keyboard though the keyboard and mouse are initially added to the DeviceInputSystem at first input.

I did find an interesting bug though.  If a mouse button is pressed but not moving when the DSM is initialized, it won't register it properly.  I've changed a bunch of the logic around and will be treating the Mouse and Touch (if applicable) as already connected devices.  Since the keyboard will constantly fire keydown events when a button is held, I don't feel like it's necessary to do the same thing with the Keyboard.

Reply:
If you are going to treat keyboard/mouse as automatically connected, a comment would be good.